app/services/oai_processing_single_service.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+require 'rest-client'
+require 'nokogiri'
+require 'traject'
+
+class OaiProcessingSingleService
+  def self.process_oai_with_marc_indexer(institution, qs, alma)
+    process_oai(institution, qs, alma, 'marc_indexer')
+  end
+
+  def self.process_oai(institution, qs, alma, ingest_tool)
+    oai_base = "https://#{alma}.alma.exlibrisgroup.com/view/oai/#{institution}/request"
+
+    log "Calling OAI with query string: #{qs}"
+    oai = RestClient.get oai_base + qs
+
+    document = Nokogiri::XML(oai)
+    marc_url = { 'marc' => "http://www.loc.gov/MARC21/slim" }
+
+    # handling of delete records
+    deleted_records = document.xpath('/oai:OAI-PMH/oai:GetRecord/oai:record[oai:header/@status="deleted"]', { 'oai' => 'http://www.openarchives.org/OAI/2.0/' })
+    suppressed_records = document.xpath("//marc:record[substring(marc:leader, 6, 1)='d']", marc_url) # gets all records with `d` in the 6th (actual) position of leader string
+    log "Found #{deleted_records.count + suppressed_records.count} delete records."
+
+    if (deleted_records.count + suppressed_records.count).positive?
+      deleted_ids = deleted_records.map { |n| n.at('header/identifier').text.split(':').last }
+      deleted_ids << suppressed_records.map { |s| s.at_xpath("marc:controlfield[@tag='001']", marc_url).text.to_i } # collects ID from controlfield 001
+      deleted_records.remove
+      suppressed_records.remove
+      puts RestClient.post "#{ENV['SOLR_URL']}/update?commit=true",
+                           "<delete><id>#{deleted_ids.join('</id><id>')}</id></delete>",
+                           content_type: :xml
+    end
+
+    # Index remaining necessary records
+    record_count = document.xpath('/oai:OAI-PMH/oai:GetRecord/oai:record', { 'oai' => 'http://www.openarchives.org/OAI/2.0/' }).count - suppressed_records.count
```

Comment:
This is one of the reasons it's breaking for me still. I made it work by replacing `suppressed_records.count` with `(suppressed_records&.count || 0)`.